### PR TITLE
bump ginkgolinter to v0.14.0 and fix findings

### PR DESCRIPTION
**Pull request by nunnatsa**
**What this PR does / why we need it**:
This PR bumps ginkgolinter to v0.14.0

This new version introduces a new rule, to prevent comparing of two different types, as the `Equal` matcher also checks types.

The PR also fixes several new findings of the ginkgolinter.

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
None
```

### CODE REVIEW

File: tests/infrastructure/prometheus.go [hunk 1]
```diff
@@ -138,14 +138,14 @@ var _ = DescribeInfra("[rfe_id:3187][crit:medium][vendor:cnv-qe@redhat.com][leve
 		}, 10*time.Second, time.Second).Should(BeTrue())
 
 		promIP := ep.Subsets[0].Addresses[0].IP
-		Expect(promIP).ToNot(Equal(0), "could not get Prometheus IP from endpoint")
+		Expect(promIP).ToNot(Equal(""), "could not get Prometheus IP from endpoint")
```
Comment (enp0s3) [inline tests/infrastructure/prometheus.go:141]: Shouldn't we use `BeEmpty()` instead of `""`?

**Review by enp0s3 — COMMENTED** [review]
@nunnatsa Thank you for taking care of this!

File: tests/infrastructure/prometheus.go (hunk 1 above)
Comment (nunnatsa) [reply tests/infrastructure/prometheus.go:141]: This a good question. I think that for strings it's more readable to use `Equal("")`. I guess it's a matter of style. Same as in a regular golang code, where some developer prefers `if s == ""`, while others will use `if len(s) == 0`. The first is more readable, and the second is with better performance. I think that in this case (unit test), I prefer readability over performance.

**Comment by jcanocan** [comment]
Good job @nunnatsa ! Thanks!
/lgtm

**Review by enp0s3 — COMMENTED** [review]
/approve

**Comment by enp0s3** [comment]
@nunnatsa Can you please check the CI failures?

**Review by 0xFelix — APPROVED** [review]
It looks good to me. Great effort, thank you! 

/lgtm

**Pull request merged**